### ISSUE TITLE
Does not work on actual Promise

### ISSUE THREAD

**Issue by 03xb7** [threepointone/react-act-examples#3]
Adapting your promise example:

```
import React, { useState, useEffect } from "react";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";

function App() {
  let [data, setData] = useState(null);
  useEffect(() => {
    fetch("/some/url").then(setData);
  }, []);
  return data;
}

it("should display fetched data", () => {
  let resolve;
  global.fetch = () => {
    return new Promise(_resolve => {
      resolve = _resolve;
    });
  };

  const el = document.createElement("div");
  act(() => {
    ReactDOM.render(<App />, el);
  });
  expect(el.innerHTML).toBe("");
  act(() => {
    resolve(42);
  });
  expect(el.innerHTML).toBe("42");
});
```

The Promise I'm using is jsdom's. I'm not using any polyfill or mock.

It both fails the test (2nd assertion) and shows the warning. My understanding is that, compared to your example, mine is not "synchronous". Is this correct ? Do you know how can I fix that without adding a synchronous promise polyfill  ?

Thanks.

**Comment by threepointone** [comment]
I've update the doc and the examples with the new async api, available in v16.9.0-alpha.0. Have a look and lemme know?

**Comment by 03xb7** [comment]
Thanks for the feedback, i'll try that.

**Comment by 03xb7** [comment]
Cool it's working. For reference, here is my updated code:

```
import React, { useState, useEffect } from "react";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";

function App() {
  let [data, setData] = useState(null);
  useEffect(() => {
    fetch("/some/url").then(setData);
  }, []);
  return data;
}

it("should display fetched data", async () => {
  let resolve;
  global.fetch = () => {
    return new Promise(_resolve => {
      resolve = _resolve;
    });
  };

  const el = document.createElement("div");
  act(() => {
    ReactDOM.render(<App />, el);
  });
  expect(el.innerHTML).toBe("");
  await act(async () => {
    resolve(42);
  });
  expect(el.innerHTML).toBe("42");
});
```

Thanks for the fix, I'll wait for 16.9 then!
